cdc/owner.go

```diff
@@ -744,13 +782,27 @@ func (o *Owner) flushChangeFeedInfos(ctx context.Context) error {
 			o.lastFlushChangefeeds = time.Now()
 		}
 	}
+
 	for _, status := range o.stoppedFeeds {
-		if status.CheckpointTs < minCheckpointTs {
-			minCheckpointTs = status.CheckpointTs
+		// if a stopped changefeed's CheckpoinTs < minGCSafePoint, means this changefeed is stagnant
+		// it should never be resume, so here we can skip it
+		if status.CheckpointTs < minGCSafePoint {
+			continue
+		}
+
+		if status.CheckpointTs < gcSafePoint {
+			gcSafePoint = status.CheckpointTs
 		}
 	}
+
+	// handle stagnant changefeed collected above
+	err = o.handleStaleChangeFeed(ctx, staleChangeFeeds, minGCSafePoint)
```

Comment:
It is possible for a changefeed to be added to `staleChangeFeeds` multiple times before it stops?

Reply:
This problem may occur, and I am working on it.

Reply:
I re-checked the entire process and found that this problem did not occur.

Reply:
I re-checked the entire process and found that this problem will not occur.